### DISCUSSION TITLE
FastAPI FileResponse

### DISCUSSION THREAD

**Discussion by SamuelePignone** [deta/cloud-docs#240] [Help]
Hi, I'm new to deta and I have a problem. I should download a file, it works locally but when I load the app on deta I get an internal server error but even with visor nothing more comes out to understand the error.

``` python 
@app.get("/test_download")
async def test_download():
    return FileResponse("file/test.glb")
```
Thank you in advance !

**Comment by jnsougata** [comment]
I guess the source of error  is the `file path` you are using when you are deploying it to deta. You have to use the path from content root.

**Reply by SamuelePignone** [reply]
I thought the same thing but trying to print the os.path.exist result returns me true.

**Reply by jnsougata** [reply]
Can you give some more code... like imports and stuff

**Reply by SamuelePignone** [reply]
Yes, of course
here's all the code

```python
from fastapi import FastAPI
from fastapi.responses import FileResponse

app = FastAPI()

@app.get('/')
async def root():
    return {"message": "Hi"}

@app.get("/test_download")
async def test_download():
    return FileResponse("file/test.glb")

@app.get("/exist")
async def exist():
    return {"msg": os.path.exists("file/test.glb")}
```

**Comment by abdelhai** [comment]
hi

1. disable visor if enabled `deta visor disable`
2. run a couple requests
3. run `deta logs` to check for system logs

**Reply by SamuelePignone** [reply]
Ok now I have discovered the error, it is in the decode of the file.

`bash
UnicodeDecodeError: 'utf-8' codec can't decode byte 0xd4 in position 8: invalid continuation byte
`

However, I still don't understand why it works locally and not in deploy

**Comment by abdelhai** [comment]
i believe this is a bug in our code. @aavshr, could we look into this?
